File: Fate.go
package main

import (
	"bufio"
	"fmt"
	"math/rand"
	"os"
	"strconv"
	"strings"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())

	reader := bufio.NewReader(os.Stdin)
	fmt.Println("Задай свой вопрос. Примечание: вопрос должен заканчиваться '?'")
	fmt.Println("Можешь указать к какому ответу больше склоняетесь. Будешь?")

	var answerPref int
	for {
		fmt.Print("> ")
		input, _ := reader.ReadString('\n')
		answer := strings.ToLower(strings.TrimSpace(input))

		if answer != "да" && answer != "нет" {
			fmt.Println("Напиши да или нет")
			continue
		}

		if answer == "да" {
			for {
				fmt.Println("Хорошо. Выбери к какому ответу склоняешься больше всего")
				fmt.Println("Нужно выбрать целое число от 1 до 99, где 0 - нет, а 100 - да")
				fmt.Print("> ")
				input, _ := reader.ReadString('\n')
				num, err := strconv.Atoi(strings.TrimSpace(input))
				if err == nil && num >= 1 && num <= 99 {
					answerPref = num
					break
				} else {
					fmt.Println("Введи целое число от 1 до 100")
				}
			}
		}
		break
	}

	// Варианты ответов
	positive := []string{"Бесспорно", "Предрешено", "Никаких сомнений", "Определённо да", "Можешь быть уверен в этом"}
	indecPositive := []string{"Мне кажется - да", "Вероятнее всего", "Хорошие перспективы", "Знаки говорят - да", "Да"}
	neutral := []string{"Пока неясно, попробуй снова", "Спроси позже", "Лучше не рассказывать", "Сейчас нельзя предсказать", "Сконцентрируйся и спроси опять"}
	negative := []string{"Даже не думай", "Мой ответ - нет", "По моим данным - нет", "Перспективы не очень хорошие", "Весьма сомнительно"}
	answers := [][]string{positive, indecPositive, neutral, negative}

	fmt.Println("Задавай вопрос")

	for {
		fmt.Print("> ")
		question, _ := reader.ReadString('\n')
		question = strings.TrimSpace(question)

		if strings.HasSuffix(question, "?") {
			category := rand.Intn(4)
			index := rand.Intn(5)

			fmt.Println(answers[category][index])
			break
		} else {
			fmt.Println("Задайте свой вопрос корректно. Примечание: вопрос должен заканчиваться '?'")
		}
	}

	fmt.Println("\nНажмите ENTER для выхода.")
	reader.ReadString('\n')
}

File: Fate.py
from random  import randint
print("Задай свой вопрос. Примечание: вопрос должен заканчиваться '?'")
print("Можешь указать к какому ответу больше склоняетесь. Будешь?")

key2 = None
key3 = None
while key2 != True:
    answer = input()
    if answer.lower() != "да" and answer.lower() != "нет":
        print("Напиши да или нет")
    else:
        if answer.lower() == "да":
            print("Хорошо. Выбери к какому ответу склоняешься больше всего")
            print("Нужно выбрать целое число от 1 до 99, где 0 - нет, а 100 - да")
            while key3 != True:
                try:
                    answer_pr = int(input())
                    if answer_pr >= 1 and answer_pr <= 99:
                        key3 = True
                        key2 = True
                    else:
                        print("Введи целое число от 1 до 100")
                except:
                    print("Введи целое число от 1 до 100")
        else:
            key2 = True

print("Задавай вопрос")
d = randint(0, 3)
l = randint(0, 4)
positive = ["Бесспорно", "Предрешено", "Никаких сомнений", "Определённо да", "Можешь быть уверен в этом"]
indecpositive = ["Мне кажется - да","Вероятнее всего","Хорошие перспективы","Знаки говорят - да","Да"]
neutral = ["Пока неясно, попробуй снова","Спроси позже","Лучше не рассказывать","Сейчас нельзя предсказать","Сконцентрируйся и спроси опять"]
negative = ["Даже не думай","Мой ответ - нет","По моим данным - нет","Перспективы не очень хорошие","Весьма сомнительно"]
s = [positive, indecpositive, neutral, negative]




#ввод
key = None
while key != 100:
    g = input()
    try:
        g = int(g)
        print("Задайте свой вопрос корректно. Примечание: вопрос должен заканчиваться '?'")
    except ValueError:
        key = True
        # вывод
        if key == True and g[-1] == "?":
            print(s[d][l])
            break
        else:
            print("Задайте свой вопрос корректно. Примечание: вопрос должен заканчиваться '?'")
input()